MapboxNavigation/RouteVoiceController.swift

```diff
@@ -209,29 +205,29 @@ open class RouteVoiceController: NSObject, AVSpeechSynthesizerDelegate {
             voiceControllerDelegate?.voiceController?(self, spokenInstructionsDidFailWith: error)
         }
         
-        var utterance: AVSpeechUtterance?
+        var utterance = AVSpeechUtterance(string: instruction.text)
         if locale?.identifier == "en-US" {
             // Alex can’t handle attributed text.
-            utterance = AVSpeechUtterance(string: instruction.text)
-            utterance!.voice = AVSpeechSynthesisVoice(identifier: AVSpeechSynthesisVoiceIdentifierAlex)
+            utterance.voice = AVSpeechSynthesisVoice(identifier: AVSpeechSynthesisVoiceIdentifierAlex)
         }
         
-        let modifiedInstruction = voiceControllerDelegate?.voiceController?(self, willSpeak: instruction, routeProgress: routeProgress) ?? instruction
-        
-        if #available(iOS 10.0, *), utterance?.voice == nil {
+        if 
+            #available(iOS 10.0, *),
+            !ignoreProgress,
+            utterance.voice == nil, 
+            let routeProgress
+        {
+            let modifiedInstruction = voiceControllerDelegate?.voiceController?(self, willSpeak: instruction, routeProgress: routeProgress) ?? instruction
+            
```

Comment:
I think you introduce an error here.

The voiceControllerDelegate should be called even if utterance.voice != nil or if iOS 10.0 isn't available. This was done before.

This function is a bit messy here. Essentially this should allow implementers to change details about the instruction before speaking it.

Reply:
I think it is correct that just for speaking an instruction, we might not necessarily require a routeProgress. In this case it might be fine to simply not create the modified instruction or set the instruction that was initially passed in this function as modifiedInstruction and then follow the regular flow.